pkg/cloud/externalconn/connectionpb/connection.go

```diff
@@ -15,9 +15,9 @@ import "github.com/cockroachdb/errors"
 // Type returns the ConnectionType of the receiver.
 func (d *ConnectionDetails) Type() ConnectionType {
 	switch d.Provider {
-	case ConnectionProvider_TypeNodelocal:
+	case ConnectionProvider_nodelocal, ConnectionProvider_s3:
```

Comment:
WHOA WHAT

Reply:
This is amazing.

Reply:
What happens when there are 10 of these?

Reply:
Heh we'll just have all the storage ones in a single case for easier readability. Let me know if I'm misunderstanding your question

Reply:
Right, I'm wondering if that will look unwieldy, but also it doesn't matter!